src/components/verification-grid/verification-grid.ts

```diff
@@ -0,0 +1,1364 @@
+    } else {
+      this.nextPage();
+    }
+  }
+
+  private async nextPage(pagedItems: number = this.gridSize): Promise<void> {
+    this.removeSubSelection();
+    this.removeDecisionHighlight();
+    this.resetSpectrogramSettings();
+
+    if (!this.getPage) {
+      throw new Error("No getPage method found.");
+    }
+
+    const nextPageIndex = this.pagedItems + pagedItems;
+    const nextPage = await this.getPage(nextPageIndex);
+    const nextVerificationModels = nextPage.map(VerificationParser.parse);
+
+    // get all verification models that are not in the undecided tiles
+    const undecidedSubjects = this.undecidedTiles.map((tile: DecisionWrapper) => tile.subject);
+    const decisionSubjects = this.decisions.map((decision: DecisionWrapper) => decision.subject);
+
+    const newVerificationModels = nextVerificationModels.filter(
+      (model: DecisionWrapper) =>
+        !undecidedSubjects.some((subject) => JSON.stringify(model.subject) === JSON.stringify(subject)) &&
+        !decisionSubjects.some((subject) => JSON.stringify(model.subject) === JSON.stringify(subject)),
+    );
+
+    const pageToRender = [...this.undecidedTiles, ...newVerificationModels];
+
+    // TODO: correct defensive programming would be incrementing the paged items each time we render an item
+    // we update the paged-items attribute last for two reasons
+    // 1. Any external applications that are watching this attribute won't
+    //    incorrectly think that the next page has been rendered before it has
+    // 2. If we fail to render the next page (e.g. the render function errors)
+    //    we want to retry rendering the page if the user presses clicks the
+    //    next page button
+    this.pagedItems = nextPageIndex;
+    this.renderVirtualPage(pageToRender);
+  }
+
+  private canNavigatePrevious(): boolean {
+    const hasHistory = this.decisions.length > 0;
+    const canNavigateBackHistory = this.historyHead < this.decisions.length;
+    return hasHistory && canNavigateBackHistory;
+  }
+
+  private canNavigateNext(): boolean {
+    const canPageNext = !(this.hiddenTiles >= this.gridSize) && !this.autoPage;
+    const canNavigateInHistory = this.isViewingHistory() && this.historyHead > this.gridSize;
+    return canPageNext || canNavigateInHistory;
+  }
+
+  //#endregion
+
+  //#region Decisions
+
+  private async catchDecision(event: DecisionEvent) {
+    // if the user is spamming a decision keyboard shortcut, it is possible for
+    // them to create two decisions for one grid tile
+    // this is a hack to prevent that
+    // TODO: this is not guaranteed to work. We should find a better solution
+    if (!this.isViewingHistory()) {
+      this.setDecisionDisabled(true);
+    }
+
+    // if the dialog box is open, we don't want to catch events
+    // because the user could accidentally create a decision by using the
+    // decision keyboard shortcuts while the help dialog is open
+    if (this.isHelpDialogOpen()) {
+      return;
+    }
```

Comment:
ok you're special casing the dialog box a lot

this and all other cases of this would be easier solved if you just unbound the keyboard events when the dialog was opened and rebound when it was closed